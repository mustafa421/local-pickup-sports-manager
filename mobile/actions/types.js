export const LOGIN_SUCCESS = "login_success";
export const LOGIN_FAIL = "login_fail";
export const LOGOUT = "logout";
export const NEW_GAMES = "new_games";
