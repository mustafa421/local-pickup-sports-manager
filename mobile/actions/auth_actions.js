import { AsyncStorage } from "react-native";
import { Facebook } from "expo";
import { LOGIN_SUCCESS, LOGIN_FAIL } from "./types";

// Helper methods

// Call backend with newly acquired account information to create account or to retrieve existing account stats
const setupUserData = async (userInfo, token, dispatch) => {
  try {
    const request = await fetch(
      "http://local-pickup-sports-manager.herokuapp.com/loginUser",
      {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(userInfo),
        json: true
      }
    );

    // check whether our API failed or not
    if (request.status !== 200) {
      dispatch({ type: LOGIN_FAIL });
    }

    const userAccountData = await request.json();
    dispatch({ type: LOGIN_SUCCESS, payload: { token, userAccountData } });
  } catch (err) {
    console.log(`Error retrieving user account from backend: ${err}`);
  }
};

// Gets facebook login token
const doFacebookLogin = async dispatch => {
  const { type, token, expires } = await Facebook.logInWithReadPermissionsAsync(
    "314718532447126",
    {
      permissions: ["public_profile", "email"]
    }
  );

  if (type === "cancel") {
    dispatch({ type: LOGIN_FAIL });
    return null;
  }

  if (type === "success") {
    try {
      await AsyncStorage.setItem("fb_token", token); // save the user's token to their device
      await AsyncStorage.setItem(
        "fb_token_expire",
        (expires * 1000).toString() // since expired token is in seconds since epoch and not milliseconds
      );
    } catch (err) {
      console.log(`Error saving facebook token ${err}`);
    }
  }

  return token;
};

export const fbLogin = () => async dispatch => {
  let token;
  let expiration;
  try {
    token = await AsyncStorage.getItem("fb_token");
    expiration = await AsyncStorage.getItem("fb_token_expire");
  } catch (err) {
    console.log(`Error getting data from storage: ${err}`);
  }

  // If the user doesn't have a token or their token has expired, have them sign in
  if (!token || parseInt(expiration, 10) < Date.now()) {
    token = await doFacebookLogin(dispatch);
    if (!token) {
      return;
    }
  }

  try {
    const response = await fetch(
      `https://graph.facebook.com/v3.2/me?fields=id,name,gender,age_range,email&access_token=${token}`
    );

    const account = await response.json();
    await setupUserData(account, token, dispatch);
  } catch (err) {
    console.log(`Error fetching user account data: ${err}`);
  }
};

export const googleLogin = () => {};
