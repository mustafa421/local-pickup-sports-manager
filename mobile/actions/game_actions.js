import { AsyncStorage } from "react-native";
import { NEW_GAMES } from "./types";

export const createGame = () => async dispatch => {};

/**
 *
 * @param {*} latitude - a number
 * @param {*} longitude - a number
 * @param {*} preferences - a fixed array representing the user's preferences
 */
export const getGames = (
  latitude,
  longitude,
  preferences
) => async dispatch => {
  try {
    const request = await fetch(
      `http://local-pickup-sports-manager.herokuapp.com/getGames?latitude=${latitude}&longitude=${longitude}&preferences=${preferences}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json"
        }
      }
    );

    if (request.status !== 200) {
      throw new Error(`Request returned ${request.status}`);
    }

    let games = await request.json();

    // Check interested and joined games
    let interested = await AsyncStorage.getItem("interestedGames");
    let joined = await AsyncStorage.getItem("joinedGames");

    if (interested) {
      interested = JSON.parse(interested);
    }

    if (joined) {
      joined = JSON.parse(joined);
    }

    // Filter the games that were joined/interested and disable their buttons
    if (games) {
      games = games.map(game => {
        const ret = Object.assign({}, game);
        if (interested && interested.includes(game.gameID)) {
          ret.interested = true;
        }

        if (joined && joined.includes(game.gameID)) {
          ret.joined = true;
        }

        return ret;
      });
    }

    dispatch({ type: NEW_GAMES, payload: games });
  } catch (err) {
    console.log(`Error when sending getGames request: ${err}`);
  }
};
