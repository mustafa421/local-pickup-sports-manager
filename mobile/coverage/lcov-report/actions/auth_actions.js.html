<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for actions/auth_actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">actions</a> auth_actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">51.43% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>18/35</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>4/12</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>17/34</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { AsyncStorage } from "react-native";
import { Facebook } from "expo";
import { LOGIN_SUCCESS, LOGIN_FAIL } from "./types";
&nbsp;
// Helper methods
&nbsp;
// Call backend with newly acquired account information to create account or to retrieve existing account stats
const setupUserData = async (userInfo, token, dispatch) =&gt; {
  try {
    const request = await fetch(
      "http://local-pickup-sports-manager.herokuapp.com/loginUser",
      {
        method: "POST",
        headers: {
          "Content-Type": "application/json"
        },
        body: JSON.stringify(userInfo),
        json: true
      }
    );
&nbsp;
    // check whether our API failed or not
    <span class="missing-if-branch" title="if path not taken" >I</span>if (request.status !== 200) {
<span class="cstat-no" title="statement not covered" >      dispatch({ type: LOGIN_FAIL });</span>
    }
&nbsp;
    const userAccountData = await request.json();
    dispatch({ type: LOGIN_SUCCESS, payload: { token, userAccountData } });
  } catch (err) {
<span class="cstat-no" title="statement not covered" >    console.log(`Error retrieving user account from backend: ${err}`);</span>
  }
};
&nbsp;
// Gets facebook login token
const doFacebookLogin = <span class="fstat-no" title="function not covered" >as</span>ync dispatch =&gt; {
  const { type, token, expires } = <span class="cstat-no" title="statement not covered" >await Facebook.logInWithReadPermissionsAsync(</span>
    "314718532447126",
    {
      permissions: ["public_profile", "email"]
    }
  );
&nbsp;
<span class="cstat-no" title="statement not covered" >  if (type === "cancel") {</span>
<span class="cstat-no" title="statement not covered" >    dispatch({ type: LOGIN_FAIL });</span>
<span class="cstat-no" title="statement not covered" >    return null;</span>
  }
&nbsp;
<span class="cstat-no" title="statement not covered" >  if (type === "success") {</span>
<span class="cstat-no" title="statement not covered" >    try {</span>
<span class="cstat-no" title="statement not covered" >      await AsyncStorage.setItem("fb_token", token); </span>// save the user's token to their device
<span class="cstat-no" title="statement not covered" >      await AsyncStorage.setItem(</span>
        "fb_token_expire",
        (expires * 1000).toString() // since expired token is in seconds since epoch and not milliseconds
      );
    } catch (err) {
<span class="cstat-no" title="statement not covered" >      console.log(`Error saving facebook token ${err}`);</span>
    }
  }
&nbsp;
<span class="cstat-no" title="statement not covered" >  return token;</span>
};
&nbsp;
export const fbLogin = () =&gt; async dispatch =&gt; {
  let token;
  let expiration;
  try {
    token = await AsyncStorage.getItem("fb_token");
    expiration = await AsyncStorage.getItem("fb_token_expire");
  } catch (err) {
<span class="cstat-no" title="statement not covered" >    console.log(`Error getting data from storage: ${err}`);</span>
  }
&nbsp;
  // If the user doesn't have a token or their token has expired, have them sign in
  <span class="missing-if-branch" title="if path not taken" >I</span>if (!token || parseInt(expiration, 10) &lt; Date.now()) {
<span class="cstat-no" title="statement not covered" >    token = await doFacebookLogin(dispatch);</span>
<span class="cstat-no" title="statement not covered" >    if (!token) {</span>
<span class="cstat-no" title="statement not covered" >      return;</span>
    }
  }
&nbsp;
  try {
    const response = await fetch(
      `https://graph.facebook.com/v3.2/me?fields=id,name,gender,age_range,email&amp;access_token=${token}`
    );
&nbsp;
    const account = await response.json();
    await setupUserData(account, token, dispatch);
  } catch (err) {
<span class="cstat-no" title="statement not covered" >    console.log(`Error fetching user account data: ${err}`);</span>
  }
};
&nbsp;
export const googleLogin = <span class="fstat-no" title="function not covered" >()</span> =&gt; {};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 16 2018 21:45:47 GMT-0600 (CST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
