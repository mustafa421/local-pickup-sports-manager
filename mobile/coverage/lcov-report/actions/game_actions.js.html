<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for actions/game_actions.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">actions</a> game_actions.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.73% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>8/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">77.78% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>7/9</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import { NEW_GAMES } from "./types";
&nbsp;
export const createGame = <span class="fstat-no" title="function not covered" >()</span> =&gt; <span class="cstat-no" title="statement not covered" ><span class="fstat-no" title="function not covered" >as</span>ync dispatch =&gt; {};</span>
&nbsp;
/**
 *
 * @param {*} location - a string representing the users location
 * TODO - Expo gives us locations via lat/long coordinates, this should
 * be added to this function
 * @param {*} preferences - a fixed array representing the user's preferences
 */
export const getGames = (location, preferences) =&gt; async dispatch =&gt; {
  try {
    // TODO - Update link after merge to cloud
    const request = await fetch(
      `http://local-pickup-sports-manager.herokuapp.com/getGames?location=${location}&amp;preferences=${preferences}`,
      {
        method: "GET",
        headers: {
          "Content-Type": "application/json"
        }
      }
    );
&nbsp;
    <span class="missing-if-branch" title="if path not taken" >I</span>if (request.status !== 200) {
<span class="cstat-no" title="statement not covered" >      throw new Error(`Request returned ${request.status}`);</span>
    }
&nbsp;
    const games = await request.json();
    dispatch({ type: NEW_GAMES, payload: games });
  } catch (err) {
<span class="cstat-no" title="statement not covered" >    console.log(`Error when sending getGames request: ${err}`);</span>
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 16 2018 21:45:47 GMT-0600 (CST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
