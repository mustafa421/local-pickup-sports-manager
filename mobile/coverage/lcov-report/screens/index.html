<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> screens
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">39.71% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>27/68</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">75.68% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>28/37</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/36</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">40.91% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>27/66</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file low" data-value="CreateGameScreen.js"><a href="CreateGameScreen.js.html">CreateGameScreen.js</a></td>
	<td data-value="25" class="pic low"><div class="chart"><div class="cover-fill" style="width: 25%;"></div><div class="cover-empty" style="width:75%;"></div></div></td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="24" class="abs low">6/24</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="20" class="abs high">18/20</td>
	<td data-value="15.38" class="pct low">15.38%</td>
	<td data-value="13" class="abs low">2/13</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="24" class="abs low">6/24</td>
	</tr>

<tr>
	<td class="file medium" data-value="MainScreen.js"><a href="MainScreen.js.html">MainScreen.js</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="12" class="abs medium">8/12</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="72.73" class="pct medium">72.73%</td>
	<td data-value="11" class="abs medium">8/11</td>
	</tr>

<tr>
	<td class="file medium" data-value="SettingsScreen.js"><a href="SettingsScreen.js.html">SettingsScreen.js</a></td>
	<td data-value="66.67" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 66%;"></div><div class="cover-empty" style="width:34%;"></div></div></td>
	<td data-value="66.67" class="pct medium">66.67%</td>
	<td data-value="9" class="abs medium">6/9</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="2" class="abs low">0/2</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	</tr>

<tr>
	<td class="file low" data-value="UpdateSettingsScreen.js"><a href="UpdateSettingsScreen.js.html">UpdateSettingsScreen.js</a></td>
	<td data-value="26.32" class="pic low"><div class="chart"><div class="cover-fill" style="width: 26%;"></div><div class="cover-empty" style="width:74%;"></div></div></td>
	<td data-value="26.32" class="pct low">26.32%</td>
	<td data-value="19" class="abs low">5/19</td>
	<td data-value="81.82" class="pct high">81.82%</td>
	<td data-value="11" class="abs high">9/11</td>
	<td data-value="22.22" class="pct low">22.22%</td>
	<td data-value="9" class="abs low">2/9</td>
	<td data-value="26.32" class="pct low">26.32%</td>
	<td data-value="19" class="abs low">5/19</td>
	</tr>

<tr>
	<td class="file medium" data-value="WelcomeScreen.js"><a href="WelcomeScreen.js.html">WelcomeScreen.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="33.33" class="pct low">33.33%</td>
	<td data-value="3" class="abs low">1/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="4" class="abs medium">2/4</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 16 2018 21:45:47 GMT-0600 (CST)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
