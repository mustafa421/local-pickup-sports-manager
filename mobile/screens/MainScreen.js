import React, { Component } from "react";
import { Permissions, Location } from "expo";
import {
  View,
  Platform,
  Alert,
  ScrollView,
  RefreshControl,
  AsyncStorage
} from "react-native";
import { Button, Text } from "react-native-elements";
import PropTypes from "prop-types";
import { connect } from "react-redux";
// eslint-disable-next-line import/no-named-as-default
import GameCard from "../components/GameCard";
import { getGames } from "../actions/game_actions";

export class MainScreen extends Component {
  static navigationOptions = ({ navigation }) => ({
    title: "Active Games",
    headerLeft: (
      <Button
        title="Settings"
        textStyle={{ color: "rgba(0, 122, 255, 1)" }}
        onPress={() => navigation.navigate("settings")}
        backgroundColor="rgba(0,0,0,0)"
      />
    ),
    headerRight: (
      <Button
        title="Create Game"
        textStyle={{ color: "rgba(0, 122, 255, 1)" }}
        onPress={() => navigation.navigate("createGameScreen")}
        backgroundColor="rgba(0,0,0,0)"
      />
    ),
    headerStyle: {
      marginTop: Platform.OS === "android" ? 24 : 0 // To prevent overlapping from header in Android devices
    }
  });

  state = { refreshing: false }; // For use with pull to refresh

  componentDidMount() {
    this.getLocationAsyncAndGetGames();
  }

  getLocationAsyncAndGetGames = async () => {
    this.setState({ refreshing: true });
    const { dispatch } = this.props;
    const { status } = await Permissions.askAsync(Permissions.LOCATION);
    if (status !== "granted") {
      Alert.alert(
        "Unable to get location",
        "Permission to get location denied, unable to get games by location, you may have to reenable permissions in settings",
        [{ text: "OK" }],
        { cancelable: false }
      );
      dispatch(getGames(null, null));
    }
    const location = await Location.getCurrentPositionAsync({});
    this.setState({ refreshing: false });
    dispatch(getGames(location.coords.latitude, location.coords.longitude));
  };

  render() {
    const { games } = this.props;
    const { refreshing } = this.state;

    /* eslint-disable react/jsx-wrap-multilines */
    return (
      <View>
        <ScrollView
          refreshControl={
            <RefreshControl
              refreshing={refreshing}
              onRefresh={this.getLocationAsyncAndGetGames}
            />
          }
        >
          {games && games.length > 0 ? (
            games.map(game => (
              // eslint-disable-next-line react/jsx-indent
              <GameCard
                key={game.gameID}
                gameID={game.gameID}
                interested={game.interested}
                joined={game.joined}
                sport={game.sport}
                title={game.title}
                skillLevel={game.skillLevel}
                duration={game.duration}
              />
            ))
          ) : (
            <Text style={{ textAlign: "center" }}>
              No games available. Be the first! Click Create Game in the top
              right corner of your screen.
            </Text>
          )}
        </ScrollView>
      </View>
    );
    /* eslint-enable react/jsx-wrap-multilines */
  }
}

const mapStateToProps = ({ game }) => ({ games: game.games });

export default connect(mapStateToProps)(MainScreen);

MainScreen.propTypes = {
  games: PropTypes.arrayOf(PropTypes.object),
  dispatch: PropTypes.func.isRequired
};

MainScreen.defaultProps = {
  games: null
};
