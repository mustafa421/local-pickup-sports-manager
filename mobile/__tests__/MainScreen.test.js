import React from "react";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";
import { fetchMock } from "fetch-mock";
import renderer from "react-test-renderer";
import { getGames } from "../actions/game_actions";
import { NEW_GAMES } from "../actions/types";
import { MainScreen } from "../screens/MainScreen";
import gameReducer from "../reducers/game_reducer";

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);

describe("Testing Main Screen functionality", () => {
  test("Main screen getGames action creator correctly adds games to store", async () => {
    const store = mockStore();

    // Set up mocked games
    const mockReturnedGames = [
      {
        title: "Basketball",
        skillLevel: "Beginner",
        duration: "2 Hours",
        gameID: 1
      },
      {
        title: "Football",
        skillLevel: "Intermediate",
        duration: "1 hour",
        gameID: 2
      },
      {
        title: "Soccer",
        skillLevel: "Expert",
        duration: "30 minutes",
        gameID: 3
      }
    ];

    // Mock the endpoint
    fetchMock.getOnce("*", mockReturnedGames);

    // Set up expected state
    const expectedActions = [
      {
        type: NEW_GAMES,
        payload: mockReturnedGames
      }
    ];

    // Call the action creator
    await store.dispatch(getGames());

    // Test that the action was received in the store
    expect(store.getActions()).toEqual(expectedActions);

    // Test our reducer correctly sets the required property
    expect(gameReducer({}, expectedActions[0])).toEqual({
      games: mockReturnedGames
    });
  });

  test("Main screen populates fixed games to match snapshopt", () => {
    const games = [
      {
        gameID: 1,
        title: "Basketball",
        skillLevel: "Beginner",
        duration: "2 Hours"
      },
      {
        gameID: 2,
        title: "Football",
        skillLevel: "Intermediate",
        duration: "1 hour"
      },
      {
        gameID: 3,
        title: "Soccer",
        skillLevel: "Expert",
        duration: "30 minutes"
      },
      {
        gameID: 4,
        title: "Hockey",
        skillLevel: "Beginner",
        duration: "1 hour"
      }
    ];

    // Pass dummy dispatch prop that returns our hardcoded games
    const tree = renderer
      .create(<MainScreen games={games} dispatch={() => false} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
