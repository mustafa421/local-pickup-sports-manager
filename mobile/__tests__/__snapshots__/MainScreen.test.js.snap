// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Main Screen functionality Main screen populates fixed games to match snapshopt 1`] = `<View />`;
